### ISSUE TITLE
About the computation of matrix exponential in iFME

### ISSUE THREAD

**Issue by LittleQuteSweetie** [cvgmi/manifold-net-dmri#1]
Hi! Thanks for sharing the code.

I have a question about the following line in `batchGLMean(M,N,w)` of fm_ops.py:
`s_i_c_pow = s_i_c**(w.view(-1))`

I guess the computation form of the geodesic here is: 
![image](https://user-images.githubusercontent.com/16911599/82887997-5c816880-9f7b-11ea-9aea-a22d11f67be8.png)

it seems like there is a matrix exponential computation of `w`*`inner_term`,
but I don't understand the `**` operation of `s_i_c` and `w`, and I think it's more likely to be :
`s_i_c_pow = torch.exp(s_i_c*(w.view(-1)))`
or is there anything wrong about the geodesic form above?

**Comment by jjbouza** [comment]
Hi, please see e.g. [here](https://www.cise.ufl.edu/~vemuri/paperphp/article.php?y=2016&i=5) for the closed form geodesic on SPN(n) with the GL invariant metric, this is what was used above. 

I'm not familiar with the expression you wrote, if you attach a reference I can probably clarify some more

**Comment by LittleQuteSweetie** [comment]
Thanks for your reply and the reference you provide, I think now I know where I was wrong before.   

The expression I wrote is from [here](https://hal.inria.fr/inria-00070743/document), which depicts a geodesic going through point Σ with tangent vector W, but the form used in `batchGLMean(M,N,w)` is the geodesic going from point M to point N.

There is no problem with the source code, and thanks again for clarifying my doubts.
